examples/miscellaneous/plot_anomaly_bench.py

```diff
@@ -0,0 +1,148 @@
+close to 1. The diagonal dashed line represents a random classification
+of outliers and inliers.
+
+
+Citation
+--------
+Shebuti Rayana (2016).  ODDS Library [http://odds.cs.stonybrook.edu].
+Stony Brook, NY: Stony Brook University, Department of Computer Science.
+
+"""
+
+from time import time
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.io import loadmat
+from sklearn.neighbors import LocalOutlierFactor
+from sklearn.ensemble import IsolationForest
+from sklearn.metrics import roc_curve, auc
+from sklearn.datasets import fetch_kddcup99, fetch_covtype
+from sklearn.preprocessing import LabelBinarizer
+
+print(__doc__)
+
+random_state = 1  # to control the random selection of anomalies in SA
+
+# datasets
+datasets = ["http", "smtp", "SA", "SF", "forestcover", "glass",
+            "vowels", "wbc", "letter"]
+
+# outlier detection models
+models = [
+    ("LOF", LocalOutlierFactor(n_neighbors=20, contamination="auto")),
+    ("IF", IsolationForest(random_state=random_state,
+                           contamination="auto")),
+]
+
+plt.figure(figsize=(5, len(datasets) * 3))
+for dataset_idx, dataset_name in enumerate(datasets):
+    plt.subplot(len(datasets), 1, dataset_idx + 1)
+    # loading and vectorization
+    print("loading data: ", str(dataset_idx + 1))
+    if dataset_name in ["http", "smtp", "SA", "SF"]:
+        dataset = fetch_kddcup99(
+            subset=dataset_name, percent10=True, random_state=random_state
+        )
+        X = dataset.data
+        y = dataset.target
+        if dataset_name == "SA":
+            idx = np.random.choice(
+                X.shape[0], int(X.shape[0]*0.1), replace=False)
+            X = X[idx, :]  # reduce the sample size
+            y = y[idx]
+
+    if dataset_name == "forestcover":
+        dataset = fetch_covtype()
+        X = dataset.data
+        y = dataset.target
+        idx = np.random.choice(
+            X.shape[0], int(X.shape[0]*0.1), replace=False)
+        X = X[idx, :]   # reduce the sample size
+        y = y[idx]
+
+        # normal data are those with attribute 2
+        # abnormal those with attribute 4
+        s = (y == 2) + (y == 4)
+        X = X[s, :]
+        y = y[s]
+        y = (y != 2).astype(int)
+
+    print("vectorizing data")
+
+    if dataset_name == "SF":
+        lb = LabelBinarizer()
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        X = np.c_[X[:, :1], x1, X[:, 2:]]
+        y = (y != b"normal.").astype(int)
+
+    if dataset_name == "SA":
+        lb = LabelBinarizer()
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        x2 = lb.fit_transform(X[:, 2].astype(str))
+        x3 = lb.fit_transform(X[:, 3].astype(str))
+        X = np.c_[X[:, :1], x1, x2, x3, X[:, 4:]]
+        y = (y != b"normal.").astype(int)
+
+    if dataset_name == "http" or dataset_name == "smtp":
+        y = (y != b"normal.").astype(int)
+
+    if dataset_name in ["glass", "vowels", "wbc", "letter"]:
```

Comment:
I am not sure about the policy regarding pushing datasets to the sklearn repo. Please fetch the datasets from [open ml](https://www.openml.org/home) using the `sklearn.datasets.fetch_openml` function (at least glass and letter are available).